stdlib/codecs.pyi

```diff
@@ -80,7 +80,7 @@ class _Encoder(Protocol):
     def __call__(self, input: str, errors: str = ..., /) -> tuple[bytes, int]: ...  # signature of Codec().encode
 
 class _Decoder(Protocol):
-    def __call__(self, input: bytes, errors: str = ..., /) -> tuple[str, int]: ...  # signature of Codec().decode
+    def __call__(self, input: memoryview, errors: str = ..., /) -> tuple[str, int]: ...  # signature of Codec().decode
```

Comment:
The input may also be `bytes`:

```
>>> import codecs
>>> codecs.lookup('utf-8').decode(b"\xc3\xb6rkki")
('örkki', 6)
```

Perhaps `_typeshed.ReadableBuffer` is better for this? It means you need to be ready to handle both `bytes` and `memoryview` as well as any other bytes-like objects. You can then do e.g. `bytes(input)` or `memoryview(input)` to convert it to a constistent type.

```suggestion
    def __call__(self, input: bytes | memoryview, errors: str = ..., /) -> tuple[str, int]: ...  # signature of Codec().decode
```

Currently `ReadableBuffer` is actually `typing_extensions.Buffer`, so code that implements decoders should probably use that, and then do e.g. `bytes(input)` or `memoryview(input)` if it needs to be converted to a consistent type.

Reply:
Makes sense, thanks!